File: Midtermlab3/MP14 – Count Keyword Occurrences/MP14_keywordCount.js
// MP14 - Count keyword occurrences

const fs = require("fs");
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question("Enter CSV file path: ", function(path) {

  rl.question("Enter keyword: ", function(keyword) {

    try {
      const data = fs.readFileSync(path, "utf8");
      const rows = data.split("\n");

      let count = 0;

      rows.forEach(row => {
        if (row.toLowerCase().includes(keyword.toLowerCase())) {
          count++;
        }
      });

      console.log(`\nKeyword '${keyword}' found in ${count} rows.`);

    } catch (error) {
      console.log("Error reading file:", error.message);
    }

    rl.close();
  });

});

File: Midtermlab3/MP14 – Count Keyword Occurrences/MP14_KeywordCount.java
// MP14 - Count keyword occurrences in CSV file

import java.io.*;
import java.util.*;

public class MP14_KeywordCount {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Ask user for dataset path
        System.out.print("Enter CSV file path: ");
        String filePath = scanner.nextLine();

        // Ask user for keyword
        System.out.print("Enter keyword to search: ");
        String keyword = scanner.nextLine().toLowerCase();

        int count = 0;

        try {
            BufferedReader br = new BufferedReader(new FileReader(filePath));
            String line;

            while ((line = br.readLine()) != null) {

                // Convert to lowercase for case-insensitive search
                String lower = line.toLowerCase();

                if (lower.contains(keyword)) {
                    count++;
                }
            }

            br.close();

            System.out.println("\nKeyword '" + keyword + "' found in " + count + " rows.");

        } catch (Exception e) {
            System.out.println("Error: " + e.getMessage());
        }

        System.out.println("Program Finished.");
        scanner.close();
    }
}